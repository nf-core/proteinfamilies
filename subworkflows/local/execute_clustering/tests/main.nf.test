nextflow_workflow {

    name "Test Subworkflow EXECUTE_CLUSTERING"
    script "../main.nf"
    workflow "EXECUTE_CLUSTERING"

    test("fasta - linclust") {

        when {
            workflow {
                """
                clustering_tool = 'linclust'

                input[0] = Channel.of([
                    [ id: 'test_linclust' ], // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/mgnifams_input_small.fa', checkIfExists: true)
                ])
                input[1] = clustering_tool
                """
            }
        }

        then {
            def lines = path(workflow.out.clusters[0][1]).readLines()
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    file(workflow.out.clusters[0][1]).name,
                    lines.size(),
                    workflow.out.versions).match() }
            )
        }
    }

    test("fasta - cluster") {

        when {
            workflow {
                """
                clustering_tool = 'cluster'

                input[0] = Channel.of([
                    [ id: 'test_cluster' ], // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/mgnifams_input_small.fa', checkIfExists: true)
                ])
                input[1] = clustering_tool
                """
            }
        }

        then {
            def lines = path(workflow.out.clusters[0][1]).readLines()
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    file(workflow.out.clusters[0][1]).name,
                    lines.size(),
                    workflow.out.versions).match() }
            )
        }
    }

    test("fasta - linclust - stub") {

        options "-stub"

        when {
            workflow {
                """
                clustering_tool = 'linclust'

                input[0] = Channel.of([
                    [ id: 'test_linclust_stub' ], // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/mgnifams_input_small.fa', checkIfExists: true)
                ])
                input[1] = clustering_tool
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("fasta - cluster - stub") {

        options "-stub"

        when {
            workflow {
                """
                clustering_tool = 'cluster'

                input[0] = Channel.of([
                    [ id: 'test_cluster_stub' ], // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/mgnifams_input_small.fa', checkIfExists: true)
                ])
                input[1] = clustering_tool
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match() }
            )
        }
    }

}
