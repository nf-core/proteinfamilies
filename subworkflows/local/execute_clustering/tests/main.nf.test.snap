{
    "fasta - cluster": {
        "content": [
            "test_cluster.tsv",
            50000,
            [
                "versions.yml:md5,3312bc6e1fa1b06e4a8ac2603deba2a4",
                "versions.yml:md5,40b242fe8ca427ffe7cd860dc5b9d7fc",
                "versions.yml:md5,9d06e5efe2356113924a205c58f0b42f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-25T08:35:58.03861131"
    },
    "fasta - linclust - stub": {
        "content": [
            {
                "0": [
                    "versions.yml:md5,3312bc6e1fa1b06e4a8ac2603deba2a4",
                    "versions.yml:md5,9d06e5efe2356113924a205c58f0b42f",
                    "versions.yml:md5,c3f8381c74eb9f0a2d549f273ba89595"
                ],
                "1": [
                    [
                        {
                            "id": "test_linclust_stub"
                        },
                        "/nf-core/test-datasets/proteinfamilies/test_data/mgnifams_input_small.fa"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test_linclust_stub"
                        },
                        "test_linclust_stub.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "clusters": [
                    [
                        {
                            "id": "test_linclust_stub"
                        },
                        "test_linclust_stub.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "seqs": [
                    [
                        {
                            "id": "test_linclust_stub"
                        },
                        "/nf-core/test-datasets/proteinfamilies/test_data/mgnifams_input_small.fa"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3312bc6e1fa1b06e4a8ac2603deba2a4",
                    "versions.yml:md5,9d06e5efe2356113924a205c58f0b42f",
                    "versions.yml:md5,c3f8381c74eb9f0a2d549f273ba89595"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-25T08:36:09.660114354"
    },
    "fasta - linclust": {
        "content": [
            "test_linclust.tsv",
            50000,
            [
                "versions.yml:md5,3312bc6e1fa1b06e4a8ac2603deba2a4",
                "versions.yml:md5,9d06e5efe2356113924a205c58f0b42f",
                "versions.yml:md5,c3f8381c74eb9f0a2d549f273ba89595"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-25T08:35:24.00287769"
    },
    "fasta - cluster - stub": {
        "content": [
            {
                "0": [
                    "versions.yml:md5,3312bc6e1fa1b06e4a8ac2603deba2a4",
                    "versions.yml:md5,40b242fe8ca427ffe7cd860dc5b9d7fc",
                    "versions.yml:md5,9d06e5efe2356113924a205c58f0b42f"
                ],
                "1": [
                    [
                        {
                            "id": "test_cluster_stub"
                        },
                        "/nf-core/test-datasets/proteinfamilies/test_data/mgnifams_input_small.fa"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test_cluster_stub"
                        },
                        "test_cluster_stub.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "clusters": [
                    [
                        {
                            "id": "test_cluster_stub"
                        },
                        "test_cluster_stub.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "seqs": [
                    [
                        {
                            "id": "test_cluster_stub"
                        },
                        "/nf-core/test-datasets/proteinfamilies/test_data/mgnifams_input_small.fa"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3312bc6e1fa1b06e4a8ac2603deba2a4",
                    "versions.yml:md5,40b242fe8ca427ffe7cd860dc5b9d7fc",
                    "versions.yml:md5,9d06e5efe2356113924a205c58f0b42f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-25T08:36:21.001725949"
    }
}